tutorials/tour/_posts/2017-02-13-by-name-parameters.md

```diff
@@ -0,0 +1,46 @@
+---
+layout: tutorial
+title: By-name Parameters
+
+disqus: true
+
+tutorial: scala-tour
+categories: tour
+num: 31
+next-page: annotations
+previous-page: operators
+---
+
+_By-name parameters_ are only evaluated when used. They are in contrast to _by-value parameters_. To make a parameter called by value, simply prepend `=>` to its type.
+
+```tut
+def goToSleep(tired: Boolean, seconds: => Int) = if (tired) Thread.sleep(seconds)
+
+def lengthOfNap = 3600 * 8
+goToSleep(tired = false, seconds = lengthOfNap)
+```
+The type of `seconds` is `=> Int` which means `seconds` is a by-name parameter. Therefore, when we call `goToSleep` it with `tired = false`, `lengthOfNap` will not be evaluated.
```

Comment:
This is not that compelling of an example. I would suggest leading with an example where avoiding unwanted evaluation is actually crucial. Computing a number of seconds is highly unlikely to be an expensive computation.